### PR TITLE
Add coverage and complete tests

### PR DESCRIPTION
**Pull request by Jeff-CCH**
Add the coverage command in README for developers to check unit test paths.

The coverage report is as below:

| File                                                | % Lines          | % Statements     | % Branches       | % Funcs         |
|-----------------------------------------------------|------------------|------------------|------------------|-----------------|
| src/Agent.sol                                       | 0.00% (0/1)      | 0.00% (0/1)      | 100.00% (0/0)    | 0.00% (0/2)     |
| src/AgentImplementation.sol                         | 100.00% (72/72)  | 98.81% (83/84)   | 85.00% (34/40)   | 100.00% (5/5)   |
| src/Router.sol                                      | 91.67% (44/48)   | 91.07% (51/56)   | 88.89% (16/18)   | 88.89% (16/18)  |
| src/callbacks/AaveV2FlashLoanCallback.sol           | 100.00% (18/18)  | 100.00% (26/26)  | 100.00% (4/4)    | 100.00% (1/1)   |
| src/callbacks/AaveV3FlashLoanCallback.sol           | 100.00% (18/18)  | 100.00% (26/26)  | 100.00% (4/4)    | 100.00% (1/1)   |
| src/callbacks/BalancerV2FlashLoanCallback.sol       | 100.00% (15/15)  | 100.00% (21/21)  | 100.00% (4/4)    | 100.00% (1/1)   |
| src/fees/AaveBorrowFeeCalculator.sol                | 100.00% (7/7)    | 100.00% (10/10)  | 100.00% (0/0)    | 100.00% (2/2)   |
| src/fees/AaveFlashLoanFeeCalculator.sol             | 100.00% (34/34)  | 100.00% (45/45)  | 83.33% (5/6)     | 100.00% (4/4)   |
| src/fees/BalancerFlashLoanFeeCalculator.sol         | 100.00% (32/32)  | 100.00% (43/43)  | 83.33% (5/6)     | 100.00% (4/4)   |
| src/fees/FeeCalculatorBase.sol                      | 100.00% (13/13)  | 100.00% (17/17)  | 100.00% (4/4)    | 60.00% (3/5)    |
| src/fees/FeeVerifier.sol                            | 87.91% (80/91)   | 89.23% (116/130) | 67.65% (23/34)   | 44.44% (4/9)    |
| src/fees/MakerDrawFeeCalculator.sol                 | 100.00% (13/13)  | 90.91% (20/22)   | 37.50% (3/8)     | 100.00% (2/2)   |
| src/fees/NativeFeeCalculator.sol                    | 100.00% (4/4)    | 100.00% (5/5)    | 100.00% (0/0)    | 100.00% (2/2)   |
| src/fees/Permit2FeeCalculator.sol                   | 100.00% (9/9)    | 92.86% (13/14)   | 50.00% (1/2)     | 100.00% (2/2)   |
| src/fees/TransferFromFeeCalculator.sol              | 100.00% (7/7)    | 100.00% (10/10)  | 100.00% (0/0)    | 100.00% (2/2)   |
| src/libraries/ApproveHelper.sol                     | 50.00% (2/4)     | 50.00% (2/4)     | 0.00% (0/4)      | 50.00% (1/2)    |
| src/libraries/LogicHash.sol                         | 0.00% (0/22)     | 0.00% (0/28)     | 100.00% (0/0)    | 0.00% (0/4)     |
| src/utilities/MakerUtility.sol                      | 56.25% (9/16)    | 60.87% (14/23)    | 75.00% (3/4)     | 100% (3/3)    |

### CODE REVIEW

**Comment by Jeff-CCH** [comment]
After I checked all the files with `Coverage Gutters`, I believe the coverage should be `100%`. The reason why we don't have 100% for some files is because:

* we test some functions with function selectors
   * eg. `OpenLockETHAndDraw` in `MakerUtility.sol`
* we test some functions only in setUp() --> I guess setUp() does not count as a test case
   * eg. `_setFeeCollector` in `Router.sol`

File: test/integration/maker/UtilityMaker.t.sol [hunk 1]
```diff
@@ -92,6 +92,29 @@ contract MakerUtilityTest is Test, MakerCommonUtils, ERC20Permit2Utils {
...
+
+        vm.prank(user);
+        vm.expectRevert(IMakerUtility.InvalidAgent.selector);
+        makerUtility.openLockETHAndDraw(
+            ethLockAmount,
+            ETH_JOIN_A,
+            DAI_JOIN,
+            bytes32(bytes(ETH_JOIN_NAME)),
+            daiDrawAmount
+        );
```
Comment (zodahu) [inline test/integration/maker/UtilityMaker.t.sol:115]: I think we can simplify the `InvalidAgent` testing by invoking the 'openLockETHAndDraw' function with some default values. For example, `makerUtility.openLockETHAndDraw(address(0),address(0)...)`

File: test/integration/maker/UtilityMaker.t.sol
Comment (zodahu) [inline test/integration/maker/UtilityMaker.t.sol:1]: Could you help rename the file to `MakerUtility.t.sol`? Thanks

File: test/integration/maker/UtilityMaker.t.sol (hunk 1 above)
Comment (Jeff-CCH) [reply test/integration/maker/UtilityMaker.t.sol]: Fixed. Please help check it again. Thanks.

File: test/integration/maker/UtilityMaker.t.sol
Comment (Jeff-CCH) [reply test/integration/maker/UtilityMaker.t.sol:1]: Fixed. Please check it again. Thanks.

**Comment by zodahu** [comment]
LGTM

**Pull request merged**